src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -42,11 +42,32 @@ public static object ConvertFromJson(string input, out ErrorRecord error)
             {
                 throw new ArgumentNullException("input");
             }
+
             error = null;
 #if CORECLR
             object obj = null;
             try
             {
+                // JsonConvert.DeserializeObject does not throw an exception when an invalid array is passed,
+                // so we need to do extra logic to identify this scenario using JArray.Parse(input).
+                if ((input.IndexOf("[", StringComparison.OrdinalIgnoreCase) >= 0) ||
+                    (input.IndexOf("]", StringComparison.OrdinalIgnoreCase) >= 0))
+                {
+                    // Remove carriage return, new-line and empty space from the string.
+                    input = input.Trim(System.Environment.NewLine.ToCharArray());
```

Comment:
Trim will only remove newline characters at the start and end. To remove all newline characters use Replace instead. And .ToCharArray() is unnecessary.